appengine-java-logging/src/test/java/com/google/apphosting/logging/JsonFormatterTest.java

```diff
@@ -85,19 +104,155 @@ public void messageIncludesStackTrace() throws Exception {
     BufferedReader reader = new BufferedReader(new StringReader(message));
     assertEquals("logger: message", reader.readLine());
     assertEquals("java.lang.Throwable: thrown", reader.readLine());
-    assertTrue(
-        reader
-            .readLine()
-            .startsWith("\tat " + getClass().getName() + ".messageIncludesStackTrace"));
+    assertTrue(reader.readLine()
+        .startsWith("\tat " + getClass().getName() + ".messageIncludesStackTrace"));
+  }
+
+  @Test
+  public void testCommonsLogging() {
+    final Instant now = Instant.now();
+    final String expectedThreadName = Thread.currentThread().getName();
+
+    // Generate Events from Commons Logging
+    new CommonsLoggingExample().run();
+
+    // Verify that event was captured by slf4j -> java.util.logging
+    String[][] expected = new String[][] {{"DEBUG", "A CommonsLogging Debug Event: "},
+        {"INFO", "A CommonsLogging Info Event: "}, {"WARNING", "A CommonsLogging Warn Event: "},
+        {"ERROR", "A CommonsLogging Error Event: "}};
+
+    List<String> events = capturingHandler.getEvents();
+    assertThat("Events", events.size(), is(expected.length));
+
+    for (int i = 0; i < events.size(); i++) {
+      String logLine = events.get(i);
+      JsonData data = new Gson().fromJson(logLine, JsonData.class);
```

Comment:
Fine as is (for a test), but the same `Gson` instance could have been used for all events.